types/src/protocol_spec.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::transaction::{TransactionInfo, TransactionInfoTrait};
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest::arbitrary::Arbitrary;
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest_derive::Arbitrary;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+pub trait ProtocolSpec:
+    Clone + Debug + Serialize + Sync + Send + for<'de> Deserialize<'de>
+{
+    const NAME: &'static str;
+    #[cfg(not(any(test, feature = "fuzzing")))]
+    type TransactionInfo: TransactionInfoTrait;
+    #[cfg(any(test, feature = "fuzzing"))]
+    type TransactionInfo: TransactionInfoTrait + Arbitrary;
+}
+
+/// Default Protocol Spec
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+pub struct DPS;
```

Comment:
Lets name this something more readable, DefaultProtocol, DiemPublicNetworksProtocol, DpnProtocol, etc.

Reply:
K. `<DefaultProcotol>` everywhere is not really my style, but I can live with it. 🙈

Reply:
I mean maybe we can solicit the team for thoughts :)